lib/inets/doc/src/httpc.xml

```diff
@@ -143,517 +72,620 @@
-      </type>
+    </datatype>
+    <datatype>
+      <name name="socket_opts"/>
+    </datatype>
+    <datatype>
+      <name name="socket_opt"/>
       <desc>
-        <p>Resets (clears) the cookie database for the specified 
-	<c>Profile</c>. If no profile is specified the default profile 
-	is used.</p>
+        <p>See the options used by
+        <seeerl marker="kernel:gen_tcp">gen_tcp(3)</seeerl> and
+        <seeerl marker="ssl:ssl">gen_tcp(3)</seeerl>
+        </p>
       </desc>
-    </func>
-    
-    <func>
-      <name since="OTP R13B04">request(Url) -> </name>
-      <name since="OTP R13B04">request(Url, Profile) -> {ok, Result} | {error, Reason}</name>
-      <fsummary>Sends a get HTTP request.</fsummary>
-      <type>
-        <v>Url = url()</v> 
-	<v>Result = {status_line(), headers(), Body} | 
-                    {status_code(), Body} | request_id()</v>
-	<v>Body = http_string() | binary()</v>
-	<v>Profile = profile() | pid()</v>
-	<d>When started <c>stand_alone</c> only the pid can be used.</d>
-	<v>Reason = term()</v>
-      </type>
+    </datatype>
+  </datatypes>
+  <datatypes>
+    <datatype_title>Types used in HTTP</datatype_title>
+    <datatype>
+      <name name="method" />
+    </datatype>
+    <datatype>
+      <name name="http_request" />
+    </datatype>
+    <datatype>
+      <name name="url" />
       <desc>
-        <p>Equivalent to <c>httpc:request(get, {Url, []}, [], [])</c>.</p>
+        <p>Syntax according to the URI definition in
+        <url href="http://www.ietf.org/rfc/rfc3986.txt">RFC 3986</url>,
+        for example <c>"http://www.erlang.org"</c></p>
+        <warning><p>Please note that httpc normalizes input URIs before internal processing
+    and special care shall be taken when the URI has percent ("%") characters. A percent
+    serves as the indicator for percent-encoded octets and it must be percent-encoded
+    as "%25" for that octet to be used as data within the URI.</p>
+    <p>For example, in order to send an HTTP GET request with the URI
+    <c>http://localhost/foo%25bar</c>, the percent character must be percent-encoded when
+    creating the request: <c>httpc:request("http://localhost/foo%2525bar").</c>
+    </p></warning>
       </desc>
-    </func>
-
-    <func>
-      <name since="OTP R13B04">request(Method, Request, HTTPOptions, Options) -></name>
-      <name since="OTP R13B04">request(Method, Request, HTTPOptions, Options, Profile) -> {ok, Result} | {ok, saved_to_file} | {error, Reason}</name>
-      
-      <fsummary>Sends an HTTP request.</fsummary>
-      <type>
-        <v>Method = method()</v>
-        <v>Request = request()</v>
-        <v>HTTPOptions = http_options()</v>
-        <v>http_options() = [http_option()]</v>
-        <v>http_option() = {timeout,         timeout()} | 
-                           {connect_timeout, timeout()} | 
-                           {ssl,             ssloptions()} | 
-                           {essl,            ssloptions()} | 
```

Comment:
I think this option should be removed from the doc and from the code. It is not documented and it seems like `ssl` and `essl` will have the same effect. I don't know the history of `essl` but now it seems redundant.

Reply:
I understand. Changing the code to remove `essl` may have further implications if some of our users were already using this option, since it would be a breaking change. I am not sure that removing `essl` from the code belongs to this PR.
I can, if it is a middle ground, remove the type spec with the `essl` option and hope that dialyzer does not complain about it.
Is this (removal of `essl` type spec) what we want for now?

Reply:
I think this PR should remove {essl,ssloptions() } from the documentation, it is mentioned as a valid option but it is not described.